### DISCUSSION TITLE
Duration and arrival rate - understanding

### DISCUSSION THREAD

**Discussion by pzur1** [artilleryio/artillery#2702] [Artillery]
Hi!
I want to run my test for all users at once. If I understand duration and arrival rate properly I need to do it like this:
`config:
  target: "http://my_target/"
  phases:
    - duration: 1
      arrivalRate: 10`

So in one second all (let's suppose 10) user will be "summoned". Am I right?
In opposite to this, scenario which looks like:
`config:
  target: "http://my_target/"
  phases:
    - duration: 10
      arrivalRate: 1`
means that for 10 seconds in each second one user will be "summoned"?
And last one:
can I put `0` in duration? Time of summoning is shorter than 1 second (?)
Regards

**Comment by bernardobridge** [comment]
Hey @pzur1,

We have some documentation with examples for this here: https://www.artillery.io/docs/reference/test-script#load-phase-examples . But yes essentially what you said is correct.

>can I put 0 in duration? Time of summoning is shorter than 1 second (?)

Could you explain what is the use case of putting `0` in duration? Currently the lowest unit is `1s`, and if you put 0 it will error.
